File: frontend/src/api/note.ts
import client from './client'

export interface Note {
  id: string
  notebook_id: string
  title: string
  content: string
  is_pinned: boolean
  created_at: string
  updated_at: string
}

export const noteApi = {
  list: async (notebookId: string): Promise<Note[]> => {
    const res = await client.get(`/notebooks/${notebookId}/notes`)
    return res.data
  },

  create: async (notebookId: string, data: { title?: string; content?: string; is_pinned?: boolean }): Promise<Note> => {
    const res = await client.post(`/notebooks/${notebookId}/notes`, data)
    return res.data
  },

  update: async (noteId: string, data: { title?: string; content?: string; is_pinned?: boolean }): Promise<Note> => {
    const res = await client.put(`/notes/${noteId}`, data)
    return res.data
  },

  remove: async (noteId: string): Promise<void> => {
    await client.delete(`/notes/${noteId}`)
  },
}

File: backend/app/schemas/note.py
"""Pydantic schemas for notes."""

from datetime import datetime

from pydantic import BaseModel


class NoteCreate(BaseModel):
    """Schema for creating a note."""

    title: str = "Untitled Note"
    content: str = ""
    is_pinned: bool = False


class NoteUpdate(BaseModel):
    """Schema for updating a note."""

    title: str | None = None
    content: str | None = None
    is_pinned: bool | None = None


class NoteResponse(BaseModel):
    """Schema for note response."""

    id: str
    notebook_id: str
    title: str
    content: str
    is_pinned: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
